soroban-auth/tests/test.rs

```diff
@@ -1 +1,96 @@
-//TODO: Write tests once auth is finalized
+#![cfg(feature = "testutils")]
```

Comment:
Why is the testutils feature required here? Looking at the tests I don't see what the testutils feature is in the auth sdk that is being used.

Reply:
`use soroban_sdk::testutils::ed25519::Sign;` requires the `testutils` feature. 

I also get this `contractimpl` error if I remove `testutils`
```
error[E0433]: failed to resolve: could not find `testutils` in `soroban_sdk`
  --> soroban-auth/tests/test.rs:44:1
   |
44 | #[contractimpl]
   | ^^^^^^^^^^^^^^^ could not find `testutils` in `soroban_sdk`
   |
   = note: this error originates in the attribute macro `contractimpl` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Reply:
Is there a better way to do this? Like including the sdk with `testutils` under `dev_dependencies`?

Reply:
Yup, should import the soroban sdk with testutils instead. Like this: https://soroban.stellar.org/docs/SDKs/rust#add-soroban-sdk-as-a-dependency

Reply:
I'm getting these errors when I remove this line and add the dev_dependency, which I've seen before when `testutils` isn't enabled.
```
error[E0277]: the trait bound `ScVec: TryFrom<(&str, &soroban_auth::Identifier)>` is not satisfied
  --> soroban-auth/tests/test.rs:10:1
   |
10 | #[contracttype]
   | ^^^^^^^^^^^^^^^ the trait `TryFrom<(&str, &soroban_auth::Identifier)>` is not implemented for `ScVec`
   |
   = help: the following other types implement trait `TryFrom<T>`:
             <ScVec as TryFrom<&std::vec::Vec<ScVal>>>
             <ScVec as TryFrom<(T0, T1)>>
             <ScVec as TryFrom<(T0, T1, T2)>>
             <ScVec as TryFrom<(T0, T1, T2, T3)>>
             <ScVec as TryFrom<(T0, T1, T2, T3, T4)>>
             <ScVec as TryFrom<(T0, T1, T2, T3, T4, T5)>>
             <ScVec as TryFrom<(T0, T1, T2, T3, T4, T5, T6)>>
             <ScVec as TryFrom<(T0, T1, T2, T3, T4, T5, T6, T7)>>
           and 6 others
   = note: required because of the requirements on the impl of `TryInto<ScVec>` for `(&str, &soroban_auth::Identifier)`
   = note: this error originates in the attribute macro `contracttype` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `ScVal: From<SignaturePayload>` is not satisfied
  --> soroban-auth/tests/test.rs:90:23
   |
90 |         signature: kp.sign(msg).unwrap().into_val(&env),
   |                       ^^^^ the trait `From<SignaturePayload>` is not implemented for `ScVal`
   |
   = help: the following other types implement trait `From<T>`:
             <ScVal as From<&()>>
             <ScVal as From<&ScStatic>>
             <ScVal as From<&bool>>
             <ScVal as From<&i32>>
             <ScVal as From<&i64>>
             <ScVal as From<&u32>>
             <ScVal as From<&u64>>
             <ScVal as From<()>>
           and 16 others
   = note: required because of the requirements on the impl of `Into<ScVal>` for `SignaturePayload`
   = note: required because of the requirements on the impl of `TryFrom<SignaturePayload>` for `ScVal`
```

Reply:
Ah, okay yeah, test utils feature gate is required then. I'll think about if there's a way to remove this requirement.